kim/utils.py

```diff
@@ -18,3 +18,15 @@ def set_creation_order(instance):
     global _creation_order
     instance._creation_order = _creation_order
     _creation_order += 1
+
+
+def attr_or_key(obj, name):
+    """attempt to use getattr to access an attribute of obj, if that fails
+    assume obj support key based look ups like a dict.
+
+    """
+
+    try:
+        return getattr(obj, name)
+    except AttributeError:
+        return obj.get(name)
```

Comment:
add another except clause here to raise a helpful error telling people we don't know WTF they're doing?